packages/template-retail-react-app/app/utils/product-utils.js

```diff
@@ -35,19 +35,103 @@ export const getDisplayVariationValues = (variationAttributes, values = {}) => {
+    if (isMaster) {
+        const variants = product?.variants || []
+        variantWithLowestPrice = variants.reduce(
+            (minVariant, variant) => {
+                const promotions = variant.productPromotions || []
+                const smallestPromotionalPrice = getSmallestValByProperty(
+                    promotions,
+                    'promotionalPrice'
+                )
+                const variantSalePrice =
+                    smallestPromotionalPrice && smallestPromotionalPrice < variant.price
+                        ? smallestPromotionalPrice
+                        : variant.price
+                return variantSalePrice < minVariant.minPrice
+                    ? {minPrice: variantSalePrice, variant}
+                    : minVariant
+            },
+            {minPrice: Infinity, variant: null}
+        )
+        currentPrice = variantWithLowestPrice?.minPrice
+    } else {
+        const promotionalPrice = getSmallestValByProperty(
+            product?.productPromotions,
+            'promotionalPrice'
+        )
+        currentPrice =
+            promotionalPrice && promotionalPrice < product?.price
+                ? promotionalPrice
+                : product?.price
+    }
+
+    // since the price is the lowest value among price books, each product will have at lease a single item tiered price at quantity 1
+    // the highest value of tieredPrices is presumptively the list price
+    const tieredPrices =
+        variantWithLowestPrice?.variant?.tieredPrices || product?.tieredPrices || []
+    const maxTieredPrice = tieredPrices?.length
+        ? Math.max(...tieredPrices.map((item) => item.price))
+        : undefined
+    const highestTieredPrice = tieredPrices.find((tier) => tier.price === maxTieredPrice)
+    const listPrice = highestTieredPrice?.price
+
+    // if a product has tieredPrices, get the tiered that has the higher closest quantity to current quantity
+    const filteredTiered = tieredPrices.filter((tiered) => tiered.quantity <= quantity)
+    const closestTieredPrice =
+        filteredTiered.length &&
+        filteredTiered.reduce((prev, curr) => {
+            return Math.abs(curr.quantity - quantity) < Math.abs(prev.quantity - quantity)
+                ? curr
+                : prev
+        })
     return {
-        basePrice,
-        discountPrice
+        currentPrice,
+        listPrice,
+        isOnSale: currentPrice < listPrice,
+        isASet,
+        isMaster,
+        // For a product, set price is the lowest price of its children, so the price should be considered a range
+        // For a master product, when it has more than 2 variants, we use the lowest priced variant, so it is  considered a range price
+        //      but for master that has one variant, it is not considered range
+        isRange: (isMaster && product?.variants?.length > 1) || isASet || false,
+        hasRepresentedProduct,
```

Comment:
What is this prop, and how it is related to pricing?

Reply:
ah good catch, it is not explicitly related to pricing, I was playing around with formatted string earlier and need to way to distinguish between PLP and PDP for different styling. Now since we are not using formatted string and the styling are consolidated for both PLP and PDP, we don't need this prop in this util technically.